app/models/king.rb

```diff
@@ -1,7 +1,8 @@
 class King < Piece
   def move_valid?(x, y)
-    return false if space_occupied?(x, y) == true
+    return false if space_occupied?(x, y) == true && opponent_color(x, y) == is_black?
```

Comment:
The `is_black` here should omit the `?` (with the `?` is asking if the column contains a value, not returning the value itself).